learn/framework/modules.md

```diff
@@ -36,3 +36,28 @@ Here is a list of some of the most used modules, but there are many more on the
 - **[pallet_staking](https://paritytech.github.io/substrate/master/pallet_staking/index.html){target=_blank}** - the Staking pallet provides functions to administer staked tokens, support rewarding, slashing, depositing, withdrawing, and so on
 
 In addition to those previously listed, other modules like [identity](https://paritytech.github.io/substrate/master/pallet_identity/index.html){target=_blank}, [smart contracts](https://paritytech.github.io/substrate/master/pallet_contracts/index.html){target=_blank}, [vesting](https://paritytech.github.io/substrate/master/pallet_vesting/index.html){target=_blank}, and many others that are freely available can speed up the development of the Appchain and, consequently, the time to market.
+
+## Custom-Made Modules {: #custom-modules }
+
+The developers creating a new module enjoy complete freedom to express any desired behavior in the core logic of the blockchain, like exposing new transactions, storing sensible information, and validating and enforcing business logic.
+
+As explained in the [Architecture](/learn/framework/architecture#client-runtime-communication) article, a module needs to be able to communicate with the core client exposing and integrating with a very specific API that allows the runtime to expose the transactions, access the storage, code and decode the information stored on-chain, and also needs to include many other required wiring codes that make the module work in the node.
+
+To improve the developer's experience when writing modules, Substrate relies heavily on [Rust macros](https://doc.rust-lang.org/book/ch19-06-macros.html){target=_blank}, which are special code instructions that generate code. The macros expand themselves by adding boilerplate code just before compiling time, allowing this way to keep up to 7X of code out of sight of the developers, facilitating them to focus on the specific functional requirements of the module instead of dealing with technicalities.
```

Comment:
```suggestion
To improve developer experience when writing modules, Substrate relies heavily on [Rust macros](https://doc.rust-lang.org/book/ch19-06-macros.html){target=_blank}, which are special code instructions that generate code. Macros expand themselves by adding boilerplate code just before compiling time, allowing them to keep up to seven times the amount of code out of sight of the developers. This allows developers to focus on the specific functional requirements of the module instead of dealing with technicalities.
```

haha i love this definition "special code instructions that generate code", while it may be accurate it doesn't really provide much context to what a macro is. I personally don't know what they are in Rust and so I feel like this definition needs to be elaborated on a little bit further for me to understand

Reply:
Without going into too much detail about macros, the point here Is that they help devs keep the code readable by hiding tons of lines. Do you think it is clearer now??

Reply:
looks good!